pkg/vm/engine/disttae/filter.go

```diff
@@ -109,24 +110,53 @@ func isSortedKey(colDef *plan.ColDef) (isPK, isSorted bool) {
 
 func getConstBytesFromExpr(exprs []*plan.Expr, colDef *plan.ColDef, proc *process.Process) ([][]byte, bool) {
 	vals := make([][]byte, len(exprs))
```

Comment:
While this may not matter at all, but this make, means people is not sure when this is called and why.  

Combined with later the pointer stuff.    Seems to me, we should pass in an extra argument to this function.   Something like a state/context, so that vals can be part of that context, and later the pointer can be an int, which is a index into some member of the state/cotnext.

Reply:
这个部分等expr复制和折叠的问题处理完之后等徐鹏或者红深再继续帮看看吧